credentialProvider/src/main/java/nl/nn/credentialprovider/ICredentialFactory.java

```diff
@@ -0,0 +1,27 @@
+/*
+   Copyright 2021 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.credentialprovider;
+
+public interface ICredentialFactory {
+
+	/**
+	 * init() returns true when the credentialFactory is properly configured and can be used.
+	 */
```

Comment:
Deze comment mag nog weg :)
Misschien in plaats daarvan een comment voor de constructor maken die aangeeft dat hij een exceptie moet gooien wanneer hij niet kan initialiseren.